OpenRA.Mods.Common/Widgets/Logic/Ingame/ReplayControlBarLogic.cs

```diff
@@ -46,17 +46,17 @@ public ReplayControlBarLogic(Widget widget, World world, OrderManager orderManag
 				var originalTimestep = world.Timestep;
 
 				var pauseButton = widget.Get<ButtonWidget>("BUTTON_PAUSE");
-				pauseButton.IsVisible = () => world.ReplayTimestep != 0 && orderManager.NetFrameNumber < replayNetTicks;
+				pauseButton.IsVisible = () => world.ReplayTimestep != 0 && !world.IsGameOver && orderManager.NetFrameNumber < replayNetTicks;
 				pauseButton.OnClick = () => world.ReplayTimestep = 0;
 
 				var playButton = widget.Get<ButtonWidget>("BUTTON_PLAY");
-				playButton.IsVisible = () => world.ReplayTimestep == 0 || orderManager.NetFrameNumber >= replayNetTicks;
+				playButton.IsVisible = () => world.ReplayTimestep == 0 || world.IsGameOver || orderManager.NetFrameNumber >= replayNetTicks;
 				playButton.OnClick = () => world.ReplayTimestep = (int)Math.Ceiling(originalTimestep * multipliers[speed]);
-				playButton.IsDisabled = () => orderManager.NetFrameNumber >= replayNetTicks;
+				playButton.IsDisabled = () => world.IsGameOver || orderManager.NetFrameNumber >= replayNetTicks;
```

Comment:
Are these `orderManager.NetFrameNumber >= replayNetTicks` checks obsoleted by `IsGameOver` now? If not, would it make sense to?

Reply:
These seem now obsolete in regards to desync or a regular game end. But all ways of ending a game should be tested, such as client connection being dropped

Reply:
https://github.com/OpenRA/OpenRA/pull/20670#issuecomment-1434472690